extras/compareResults.R

```diff
@@ -1,73 +1,74 @@
+#' Comparing DQD results
+#' 
+#' Use:
+#' compare_results(file_old, file_new, saving_dir)
+
 library("dplyr")
-library("jsonlite")
 library("ggplot2")
-library("plotly")
-
-# Directories
-setwd('/my/path/to/dqd/extra/scripts/')
-file_old <- "json/results/folder/file1.json"
-file_new <- "json/results/folder/file2.json"
-saving_dir <- "results_compare/"
-dir.create(file.path(getwd(), saving_dir), showWarnings = FALSE)
-
-
-# List all differences
-# ... between OLD
-result_old <- jsonlite::fromJSON(file_old)
-check_results_old <- tibble(result_old$CheckResults)
-
-# ... and NEW
-result_new <- jsonlite::fromJSON(file_new)
-check_results_new <- tibble(result_new$CheckResults)
+# Other packages used: jsonlite, plotly
 
-# ... only keep the different
-combined_results <- check_results_old %>%
-  left_join(check_results_new,
-            by=c("CHECK_NAME", "CDM_TABLE_NAME", "CDM_FIELD_NAME",
-                 "CONCEPT_ID", "UNIT_CONCEPT_ID"),
-            suffix=c(".old", ".new")) %>%
-  filter(PCT_VIOLATED_ROWS.old != PCT_VIOLATED_ROWS.new)
+compare_results <- function(jsonPath.old, jsonPath.new, saving_dir){
 
-# Save as csv
-write.csv(combined_results, file=file.path(getwd(), saving_dir, "different_checks.csv"))
-
-# If no differences: add mock point
-if(nrow(combined_results)==0){
-  stop("No differences found.")
+  # List all differences
+  # ... between OLD
+  result_old <- jsonlite::fromJSON(file_old)
+  check_results_old <- tibble(result_old$CheckResults)
+  
+  # ... and NEW
+  result_new <- jsonlite::fromJSON(file_new)
+  check_results_new <- tibble(result_new$CheckResults)
+  
+  # ... only keep the different
+  combined_results <- check_results_old %>%
+    left_join(check_results_new,
+              by=c("CHECK_NAME", "CDM_TABLE_NAME", "CDM_FIELD_NAME",
+                   "CONCEPT_ID", "UNIT_CONCEPT_ID"),
+              suffix=c(".old", ".new")) %>%
+    filter(PCT_VIOLATED_ROWS.old != PCT_VIOLATED_ROWS.new)
+  
+  # Save as csv
+  saving_name <- file.path(saving_dir, paste("compare_dqd", Sys.Date(), sep="_"))
+  dir.create(file.path(saving_dir), showWarnings = FALSE)
+  write.csv(combined_results, file=paste(saving_name, ".csv", sep=""))
+  
+  # If no differences: add mock point
+  if(nrow(combined_results)==0){
+    stop("No differences found.")
+  }
```

Comment:
This block can be moved before the saving as csv (line 29). If `combined_results` is empty, it will just output an empty csv table.
```suggestion
  # No difference found, exit function
  if(nrow(combined_results)==0){
    stop("No differences found.")
  }
```